src/DriftKernelAdapter.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Antidot\React;
+
+use Antidot\Application\Http\Application;
+use Drift\Console\OutputPrinter;
+use Drift\Server\Adapter\KernelAdapter;
+use Drift\Server\Context\ServerContext;
+use Drift\Server\Mime\MimeTypeChecker;
+use Psr\Container\ContainerInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use React\EventLoop\LoopInterface;
+use React\Filesystem\FilesystemInterface;
+use React\Promise\PromiseInterface;
+use function React\Promise\resolve;
+
+final class DriftKernelAdapter implements KernelAdapter
+{
+    private ServerContext $serverContext;
+    private MimeTypeChecker $mimeTypeChecker;
+    private string $rootPath;
+    private ContainerInterface $container;
+    private ReactApplication $application;
+    /** @var object|FilesystemInterface|null  */
+    private ?object $filesystem = null;
+
+    public function __construct(
+        ServerContext $serverContext,
+        MimeTypeChecker $mimeTypeChecker,
+        string $rootPath,
+        ?object $filesystem
```

Comment:
This should be strict `?FilesystemInterface $filesystem`. It doesn't matter if the dependency is not there, the contract should be working as well.

Reply:
OOps. Nopes, sorry, this is not the method I was talking about :(